More algo unification

papampi opened the pull request:
Unification of NEOSCRYPT, LYRA2V2, SKEIN, ZPOOL, ...

papampi committed:
Update 0miner

More algo unification: NEOSCRYPT, LYRA2V2, SKEIN, ZPOOL, ...

papampi committed:
Update 0miner

LuKePicci commented on 0miner:
```diff
@@ -462,37 +375,20 @@ then
 
 if [[ $COIN == NICE-SKUNKHASH ]]
 then
   HCD=${NVOC}/miners/"$SKUNK_MINER"/ccminer
```
Why not `HCD="${NVOC}/miners/$SKUNK_MINER/ccminer"` ? Why quoting this one and a few other only?

papampi replied:
Will run shellcheck on it when all done.

papampi committed:
Update 0miner

MPH Algos

papampi committed:
More Algo miners

Cryptonight, Cryptonightv7, Timetravel10, X16R, X11ghost/sib, Lyra2z

papampi committed:
Update 0miner

papampi commented:
Not sure if we should do this or not:
https://github.com/papampi/nvOC_by_fullzero_Community_Release/blob/5c96032def8449672b167a462dea56cb6740815e/0miner#L274-L278

LuKePicci commented:
( never used stack-miner, but I think those files should be into its miner folder. What's their content?

papampi committed:
Update 0miner

Added all the rest of algos
Run shellcheck on it and fixed all warnings

papampi commented:
When xmr-stack run for first time it create them in home folder no matter where you start the miner from

papampi commented:
1bash and 3main needs integration with latest 0miner, will wait for PR #59

papampi committed:
Update 0miner

Change ZM to DSTM for better identification

LuKePicci commented:
> When xmr-stack run for first time it create them in home folder no matter where you start the miner from

So that should be `~/` instead of `${NVOC}/`

papampi committed:
Update 0miner

papampi commented:
My question is should we remove that nvidia.txt file and make miner create it every time it starts or add an info line in 1bash so that users remove it if they add/remove cards

LuKePicci commented:
I don't know what is written into that file, so I can't say. If it contains any hint about the number of cards you could automatically remove it if that number do not match the detected GPUs count.

papampi commented:
~/nvidia.txt from my test rig
I'm afraid removing/creating it every time will make miner start slower


```
/*
 * GPU configuration. You should play around with threads and blocks as the fastest settings will vary.
 * index         - GPU index number usually starts from 0.
 * threads       - Number of GPU threads (nothing to do with CPU threads).
 * blocks        - Number of GPU blocks (nothing to do with CPU threads).
 * bfactor       - Enables running the Cryptonight kernel in smaller pieces.
 *                 Increase if you want to reduce GPU lag. Recommended setting on GUI systems - 8
 * bsleep        - Insert a delay of X microseconds between kernel launches.
 *                 Increase if you want to reduce GPU lag. Recommended setting on GUI systems - 100
 * affine_to_cpu - This will affine the thread to a CPU. This can make a GPU miner play along nicer with a CPU miner.
 * sync_mode     - method used to synchronize the device
 *                 documentation: http://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__DEVICE.html#group__CUDART__DEVICE_1g69e73c7dda3fc05306ae7c811a690fac
 *                 0 = cudaDeviceScheduleAuto
 *                 1 = cudaDeviceScheduleSpin - create a high load on one cpu thread per gpu
 *                 2 = cudaDeviceScheduleYield
 *                 3 = cudaDeviceScheduleBlockingSync (default)
 *
 * On the first run the miner will look at your system and suggest a basic configuration that will work,
 * you can try to tweak it from there to get the best performance.
 *
 * A filled out configuration should look like this:
 * "gpu_threads_conf" :
 * [
 *     { "index" : 0, "threads" : 17, "blocks" : 60, "bfactor" : 0, "bsleep" :  0, 
 *       "affine_to_cpu" : false, "sync_mode" : 3,
 *     },
 * ],
 * If you do not wish to mine with your nVidia GPU(s) then use:
 * "gpu_threads_conf" :
 * null,
 */

"gpu_threads_conf" :
[
  // gpu: GeForce GTX 1070 architecture: 61
  //      memory: 7712/8119 MiB
  //      smx: 15
  { "index" : 0,
    "threads" : 44, "blocks" : 45,
    "bfactor" : 0, "bsleep" :  0,
    "affine_to_cpu" : false, "sync_mode" : 3,
  },
  // gpu: P106-100 architecture: 61
  //      memory: 5996/6080 MiB
  //      smx: 10
  { "index" : 1,
    "threads" : 32, "blocks" : 30,
    "bfactor" : 0, "bsleep" :  0,
    "affine_to_cpu" : false, "sync_mode" : 3,
  },
  // gpu: P106-100 architecture: 61
  //      memory: 5996/6080 MiB
  //      smx: 10
  { "index" : 2,
    "threads" : 32, "blocks" : 30,
    "bfactor" : 0, "bsleep" :  0,
    "affine_to_cpu" : false, "sync_mode" : 3,
  },

],

```

papampi committed:
Update 0miner

papampi committed:
Update 0miner

LuKePicci commented:
`grep -c "// gpu:" ~/nvidia.txt` should work for counting, usin jq is not an option here since it is not in valid json

papampi committed:
Update 0miner

CryptonightV7 GPU count check

papampi committed:
Update 1bash

Algo unified miners selection

papampi committed:
Update 3main

More coin/algo identification

papampi committed:
Update 0miner

papampi committed:
Update 3main

papampi committed:
Merge branch '19-2.1' into papampi-0miner_unification

papampi committed:
Update 0miner

Removed not needed mph and zpool worker variables

papampi committed:
Update 0miner

papampi committed:
Update 0miner

New approach ...

papampi committed:
Update 0miner

papampi committed:
Update 1bash

papampi committed:
Update 0miner

papampi committed:
Update 1bash

papampi committed:
Update 0miner

papampi committed:
Update 1bash

papampi committed:
Update 0miner

papampi commented:
I think its done ...

sizzlephizzle reviewed (commented):
I can find another 26 lines that can be removed
I can't modify this though. Can you include me on contributor list please?
UMINER="_MINER"
xminer=$COIN$UMINER

HCD=${NVOC}/miners/"${!xminer}"/ccminer

"$LAUNCH" "$HCD" "$UCCALGO" -o stratum+tcp://"${!xpool}":"${!xport}" -u "${!xadd}""$WALLET_ADDRESS_FORMAT""${!xwrk}" -p "$MINER_PWD" -i "${!xintensity}" "$CCMINER_OPTS"

papampi committed:
xminer and xwallet

papampi commented:
@sizzlephizzle you are already in contributor list
Added xminer and xwallet

papampi committed:
Update 1bash

sizzlephizzle commented on 0miner:
```diff
@@ -69,430 +72,255 @@ UPORT="_PORT"
+  elif [[ $ALGO == KECCAK ]]
+  then
+    WALLET_ADDRESS_FORMAT="$KECCAK_WALLET_FORMAT"
+    HCD=${NVOC}/miners/"$XEVAN_MINER"/ccminer
```
Xevan for Keccak? Not sure if this is right or wrong

papampi replied:
Was a mistake and fixed with xminer...

papampi committed:
Update 0miner

papampi committed:
Update 1bash

sizzlephizzle commented:
Had an idea - there are several coins for ccminer where the ALGO is the same text as -a algo but the wrong case (Capitals, not lowercase).

Googled: 
sed -e 's/\(.*\)/\L\1/' $ALGO $algo

Should convert
elif [[ $ALGO == GROESTL ]]
UCCALGO="-a groestl"
to something like
if [[ $ALGO == X16R || $ALGO == SIB || $ALGO == LYRA2Z || $ALGO == SKUNK || $ALGO == KECCAK || $ALGO == SKEIN || $ALGO == NEOSCRYPT || $ALGO == LYRA2V2 || $ALGO == TIMETRAVEL10 || $ALGO == PHI || $ALGO == LBRY || $ALGO == GROESTL || $ALGO == SIA || $ALGO == X13 ]]
sed -e 's/\(.*\)/\L\1/' $ALGO $algo
UCCALGO="-a $algo"
$LAUNCH
fi

github keeps saying I need to be in a branch, but I can't get it working

papampi committed:
Update 0miner

papampi committed:
Update 3main

papampi commented:
Open the branch and edit from there ...
https://github.com/papampi/nvOC_by_fullzero_Community_Release/tree/papampi-0miner_unification

sizzlephizzle commented:
So 0miner is now 213 lines.....
Pretty good going!
The sed command wont convert $ALGO == DMD-GR to dmd_gr will it? Same goes for $ALGO == MYR-GR

Unless we have a policy of naming the $ALGO for ccminer to whatever the -a switch is? That would need DMD-GR renaming to DMD_GR in 3main etc.
This is a discussion point. Welcome your feedback.

papampi commented:
I think they should be "-a dmd-gr" and "-a myr-gr", thats why I renamed them from _ to -

papampi commented:
sed -e 's/_/-/g' will convert "_" to "-" but why when we can have the correct one in 3main

papampi committed:
Update 1bash

More MPH algo

sizzlephizzle commented:
That's pretty much everything we could possibly change. I think we're done unless you can spot anything else?

papampi committed:
Update 3main

papampi commented:
Added some more MPH_ALGO, 
Should check if all ZPOOL_ALGO are included in 3main algo identification

LuKePicci commented on 3main:
```diff
@@ -329,8 +329,44 @@ then
+elif [[ $COIN == ONION ]]
+then
+  ALGO=X13  
+elif [[ $COIN == SALFTER_NICEHASH_PROFIT_SWITCHING || $COIN == SALFTER_MPH_PROFIT_SWITCHING || $COIN == SALFTER_PROGRAMATIC_SWITCHING ]]
```
This block must be removed

papampi replied:
The ONION block or Salfter block ?

LuKePicci replied:
The Salfter block - LOL
Github show three previous lines of code context wrt that one I commented on. It was already removed in master, maybe the merge will do it automatically.

papampi replied:
Done..

papampi committed:
Update 3main

papampi committed:
Update 1bash

papampi committed:
Update 3main

ALGO=SALFTER removed

papampi committed:
Update 0miner

papampi committed:
Update 1bash

papampi committed:
Update 3main

papampi commented on 0miner:
```diff
@@ -58,441 +61,150 @@ if ps ax | grep SCREEN | grep -v cpuminer | grep -q miner ; then
+  echo ""
+  echo "COIN/ALGO not found, Check your settings"
+  echo "Miner not started, Stopping watchdog, tempcontrol, ..."
+  pkill -e screen
```
@LuKePicci 
Do we have any screens that should stay alive if miner not started?

LuKePicci replied:
Good point, for sure if tempcontrol has taken control of fans, maybe a good idea to let it run, so it slows down fans as long as gpu are getting cold. However there is probably no reason to kill anything here, just logging the event in the stdout/telegram would be enough.

papampi replied:
Then lets kill watchdog only

papampi commented on 1bash:
```diff
@@ -364,25 +366,27 @@ OVERCLOCK_MODE="GLOBAL"     # NO, GLOBAL, GPU_SPECIFIC, ALGO_SPECIFIC
 
```
@LuKePicci 
Can you please add info in 1bash for new OC settings?
I think there are missing info about the 2 new OFFSET modes...

LuKePicci replied:
There were, but as @leenoox pointed out they're not so simple to understand for newbies, I'd rather (re)add in a future parallel 1bash template for pro users, meanwhile let's keep it simple.

papampi replied:
I remember our talks with leenoox, but I also think since we already have the codes, let's give pro users chance to use em, there are already people asking for those options in discord.
You can add a Pro OC settings so beginners dont mess with those...

papampi committed:
Update 1bash

Need info on ALGO_SPECIFIC_with_GPU_OFFSET and GLOBAL_with_GPU_OFFSET

papampi committed:
Update 0miner

papampi committed:
Update 1bash

#        - Pro Settings are not for the faint-hearted. Don't use if you don't know what they are. 
lol ... Good now?

papampi committed:
Update 0miner

Changed timetravle10 to bitcore

papampi committed:
Update 1bash

papampi committed:
Update 0miner

papampi commented on 0miner:
```diff
@@ -58,441 +60,150 @@ if ps ax | grep SCREEN | grep -v cpuminer | grep -q miner ; then
+elif [[ $ALGO == X16R || $ALGO == SIB || $ALGO == LYRA2Z || $ALGO == SKUNK || $ALGO == KECCAK || $ALGO == SKEIN || $ALGO == NEOSCRYPT || $ALGO == LYRA2V2 || $ALGO == BITCORE || $ALGO == PHI || $ALGO == LBRY || $ALGO == DMD-GR || $ALGO == GROESTL || $ALGO == MYR-GR || $ALGO == DECRED || $ALGO == SIA || $ALGO == X13 ]]
+then
+  UCCALGO="-a $(echo "$ALGO" | sed -e 's/\(.*\)/\L\1/')"
+  HCD="${NVOC}"/miners/"${!xminer}"/ccminer
```
@LuKePicci 
shellcheck asking to use this code instead, you know how to?
```
  UCCALGO="-a $(echo "$ALGO" | sed -e 's/\(.*\)/\L\1/')"
                ^-- SC2001: See if you can use ${variable//search/replace} instead.
```

LuKePicci replied:
if you only need to get lowecase version of the variable ALGO you can just use ${ALGO,,}

papampi committed:
Update 1bash

Set default Algo specific OC to 0 with a hint for some 1060 cards

papampi commented:
If PL is less than minimum allowed card limit it will set it to minimum allowed by manufacturer, but not tested if its set to 0, will test later ...

papampi committed:
Update 3main

Timetravel10 to bitcore

papampi committed:
Update 3main

papampi committed:
Update 0miner

papampi committed:
Update 1bash

Power limit 0 gives error in temp control, keep value at default.

papampi commented:
Isn't better to merge both lowercase and UPPERCASE OC values into single one for all ?

sizzlephizzle commented:
I don't see it as an issue. It's just a different way to call the same
variable. One single variable called differently. I haven't seen it in
action so I'm not in a good place to comment, but if Luca says it will
work, it's better than calling sed to add to processing power/time on the
run through (In my opinion)

On Thu, May 24, 2018 at 9:33 AM, Payam Nab <notifications@github.com> wrote:

> Reopened #60
> <https://github.com/papampi/nvOC_by_fullzero_Community_Release/pull/60>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/papampi/nvOC_by_fullzero_Community_Release/pull/60#event-1643647988>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AiaWraFFODmZoQV9GE41KIsEv9gKRo0wks5t1nBqgaJpZM4UDSCq>
> .
>

LuKePicci commented:
There are problems deriving from the different algo selection mechanisms used by salfter's switchers.
If WTM_SWITCHING switch to an algo for which there is no OC specific settings it falls back to globals. Salfter's switchers support only a small set of algos and require all of them to be set in order to work as expected. They perform algo selection at runtime outside the control of 3main and are not updated in case of switching without restarting 3main. They can be merged, I but I already evaluated at that time not to do it because will create potential confusion because one could think a certain algo is supported by Salfter's switchers while it's not.

papampi commented:
Then if there is nothing more to add merge this and talk about unifying all algos later ...

LuKePicci reviewed (approved):
Agree
